libs/math/tests/unit/fixed_point/fixed_point.cpp

```diff
@@ -810,6 +819,42 @@ TYPED_TEST(BasicArithmeticTest, Division)
   TypeParam::StateClear();
   EXPECT_TRUE(TypeParam::IsNaN(zero / zero));
   EXPECT_TRUE(TypeParam::IsStateNaN());
+
+  // Test associativity with primitives
+  EXPECT_EQ(three / 2.0, 3.0 / two);
+}
+
+template <typename T>
+class BitOperationsTest : public ::testing::Test
+{
+};
+TYPED_TEST_CASE(BitOperationsTest, FixedPointTypes);
+TYPED_TEST(BitOperationsTest, ShiftLeft)
+{
+  // Positive
+  TypeParam one{1};
+  TypeParam two{2};
+  TypeParam x{1.6519711627625};
```

Comment:
initialisation with a float, not a string constructor - is it intended?

Reply:
yes, for checking associativity and operations it's fine :)

Reply:
The float constructor is still there for now, but as it's not safe to use in library code I think in the long run we'll have to disable it. So I think it would be better to use the string constructor here.